.github/workflows/firefox-ios-uri-update.yml

```diff
@@ -0,0 +1,87 @@
+        default: 'main'
+jobs:
+  build:
+    runs-on: ubuntu-latest # using ubuntu as a lightweight environment
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+        token: ${{ secrets.GITHUB_TOKEN }}
+        ref: ${{ github.event.inputs.branchName }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.9'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pandas requests
+
+    - name: Run URI update script
+      run: python taskcluster/scripts/uri_update.py
+
+    - name: Check for changes
+      id: git-check
+      run: |
+        if git diff --quiet; then
+          echo "No changes detected, skipping PR creation."
+          echo "changes_detected=false" >> $GITHUB_ENV
+        else
+          echo "Changes detected, proceeding to create PR."
+          git diff
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email '41898282+github-actions[bot]@users.noreply.github.com'
+          git commit --allow-empty -m "empty commit" # add empty commit so when squashed the history with show PR title
+          git add firefox-ios/Shared/Extensions/URLExtensions.swift
+          echo "changes_detected=true" >> $GITHUB_ENV
+          datetime=$(date +%Y%m%d)
+          echo "date=$datetime" >> $GITHUB_ENV
+        fi
+
+    - name: Determine PR Version Number
+      id: versioning
+      run: |
+          
+          git fetch --all
+          # get highest current release/v branch, strip the numbers, and increment by one to get the current
+          # release version target based on that state 'main'
+          # if no release branch is found, default to an empty string to avoid exiting script
+          highest_version=$(git branch -r | \
+            { grep 'release/v' | \
+            awk -F'/' '{print $NF}' | \
+            sort -Vr | \
+            head -n 1 | \
+            grep -o '[0-9]*' || echo ""; })
+
+          if [[ ! -z "$highest_version" ]]; then
+            next_version=$((highest_version + 1))
+          else
+            echo "Couldn't determine current release version. Defaulting to [vXXX]"
+            next_version=XXX
+          fi
+
+          echo "Next version is v$next_version"
+          echo "next_version=$next_version" >> $GITHUB_ENV
+
+    - name: Create Pull Request
+      if: env.changes_detected == 'true'
+      uses: peter-evans/create-pull-request@v5
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        commit-message: Automated URI update on ${{ github.event.inputs.branchName || 'main' }} ${{ env.date }}
+        author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
+        committer: GitHub <noreply@github.com>
+        title: Refactor [v${{ env.next_version }}] Automated URI update on ${{ github.event.inputs.branchName || 'main' }} ${{ env.date }}
```

Comment:
This is nice to get the next version in the PR name automatically! Some other Github actions would also benefit from this. Any way we could share this with other actions easily?

Reply:
Sure, I don't think it would be tough.  it's just some bash to get the highest numbered `release/vXXX` branch and increment it by one. 

I created a seperete step in the `Github` action for this the other workflows could add, `Determine PR Version Number`

Reply:
Oh that's great. That would be useful for the AS automatic PR too. We can open a ticket to add it as it will remove one manual step when reviewing those PRs. ``

Reply:
We can use it in `firefox-ios-update-effective-tld-names-file.yml` and `firefox-ios-import-strings.yml` as well 👀

Reply:
Jira ticket for that: https://mozilla-hub.atlassian.net/browse/MTE-2296